splink/internals/comparison_level_library.py

```diff
@@ -742,6 +742,66 @@ def create_sql(self, sql_dialect: SplinkDialect) -> str:
 
     def create_label_for_charts(self) -> str:
         return f"Distance less than {self.km_threshold}km"
+    
+class ArrayStringDistanceLevel(ComparisonLevelCreator):
+    def __init__(self, col_name: str | ColumnExpression, distance_threshold: int, distance_function: str):
+        """Represents a comparison level based around the distance between
+        arrays
+
+        Args:
+            col_name (str): Input column name
+            distance_threshold (int): the maximum distance between string 
+                elements in the arrays for this comparison level.
+            distance_function (str): Distance function name to calculate 
+                pair-wise between arrays
+        """
+
+        self.col_expression = ColumnExpression.instantiate_if_str(col_name)
+        self.distance_threshold = validate_numeric_parameter(
+            lower_bound=0,
+            upper_bound=float("inf"),
+            parameter_value=distance_threshold,
+            level_name=self.__class__.__name__,
+            parameter_name="distance_threshold",
+        )
+        self.distance_function = validate_categorical_parameter(
+            allowed_values=["levenshtein", "damerau_levenshtein", "jaro_winkler", "jaro"],
+            parameter_value=distance_function,
+            level_name=self.__class__.__name__,
+            parameter_name="distance_function"
+        )
```

Comment:
I think it's an open question whether it is better for the user to define any function name they want (as in `DistanceFunctionLevel`) or only to have certain options (but it gets auto-transpiled).